webapp/src/components/preview-movies/style/row/TopRow.tsx

```diff
@@ -1,54 +1,119 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { Production } from '../../../../api/types/MovieDbTypes';
 import RowLoading from '../../../general/loading/RowLoading';
-import '../../../../../assets/scss/components/row.scss';
-import '../../../../../assets/scss/components/arrow.scss';
-import '../../../../../assets/scss/components/top-row.scss';
-import Player from '../../../general/streaming/movie/Player';
-import Arrow from './Arrow';
-import Poster from '../image/Poster';
+import '../../../../../assets/scss/components/style/row/row.scss';
+import '../../../../../assets/scss/components/style/arrow/arrow.scss';
+import '../../../../../assets/scss/components/style/row/top-row.scss';
+import Arrow from '../arrow/Arrow';
+import Poster from '../poster/Poster';
 
 type Props = {
   title?: string,
   movieList: Production[],
   isDataLoading?: boolean
 };
 
+const SLIDER_TIMING: number = 8000;
+
 function TopRow({ movieList, isDataLoading }: Props) {
-  function hundleClickArrowRight() {
-    const slider = document.getElementById('top_posters');
-    slider.scrollLeft += window.innerWidth;
+  const slider = useRef<HTMLDivElement>(null);
+
+  const [currentPoster, setCurrentPoster] = useState(0);
+
+  const sliderInterval = useRef<NodeJS.Timeout>(setInterval(() => {
+    goNextPoster();
+  }, SLIDER_TIMING));
+
+  function createSliderInterval() {
+    return sliderInterval.current;
+  }
+
```

Comment:
il faut mettre à jour la ref à chaque fois que tu crées l'interval.
```suggestion
  function createSliderInterval() {
    if(sliderInterval.current) {
        stopSliderInterval();
    }
    sliderInterval.current = setInterval(() => goNextPoster(), SLIDER_TIMING);
  }
```

Reply:
Ah d'accord, j'avais pensé à ça mais je pensais que la ref avait toujours comme valeur le TimeOut que je lui ai donné lors de sa création. Je pensais que quand on le stop ça arrête le timer sans supprimer l'objet.

Reply:
il faut qu'on revoit ça ensemble, tu semble ne pas bien comprendre comment ça fonctionne